### PR TITLE
Support Xcode 8 beta 6 and Preview 6

### PR DESCRIPTION
**Pull request by ikesyo**
~~The tests are not updated yet. Feel free to take over the work here.~~ 🙏

That is done by https://github.com/ikesyo/Nimble/pull/1 ✨

### CODE REVIEW

File: Sources/Nimble/Adapters/NMBObjCMatcher.swift [hunk 1]
```diff
@@ -2,8 +2,8 @@ import Foundation
 
 #if _runtime(_ObjC)
 
-public typealias MatcherBlock = (actualExpression: Expression<NSObject>, failureMessage: FailureMessage) -> Bool
-public typealias FullMatcherBlock = (actualExpression: Expression<NSObject>, failureMessage: FailureMessage, shouldNotMatch: Bool) -> Bool
+public typealias MatcherBlock = (_ actualExpression: Expression<NSObject>, _ failureMessage: FailureMessage) -> Bool
```
Comment (NachoSoto) [inline Sources/Nimble/Adapters/NMBObjCMatcher.swift]: This change makes me sad.

File: Sources/Nimble/DSL+Wait.swift [hunk 2]
```diff
@@ -15,8 +15,8 @@ internal class NMBWait: NSObject {
         timeout: TimeInterval,
         file: FileString = #file,
         line: UInt = #line,
-        action: (() -> Void) -> Void) -> Void {
-            return throwableUntil(timeout: timeout, file: file, line: line) { (done: () -> Void) throws -> Void in
+        action: @escaping (@escaping () -> Void) -> Void) -> Void {
+            return throwableUntil(timeout: timeout, file: file, line: line) { done in
                 action() { done() }
```
Comment (NachoSoto) [inline Sources/Nimble/DSL+Wait.swift]: I think this could pass `done`  directly instead of creating another closure?

**Comment by norio-nomura** [comment]
@ikesyo I pushed PR against your branch.
That is still failing to build tests on:
- Removal of implicit bridging conversion
- Changes of operator functions.

**Comment by ikesyo** [comment]
All green 💚

**Pull request merged**